app/src/test/java/org/simple/clinic/appupdate/CheckAppUpdateAvailabilityTest.kt

```diff
@@ -0,0 +1,75 @@
+package org.simple.clinic.appupdate
+
+import android.app.Application
+import android.app.PendingIntent
+import com.google.android.play.core.appupdate.AppUpdateInfo
+import com.google.android.play.core.install.model.InstallStatus
+import com.google.android.play.core.install.model.UpdateAvailability
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(JUnitParamsRunner::class)
+class CheckAppUpdateAvailabilityTest {
+
+  private val config = mock<AppUpdateConfig>()
+
+  lateinit var checkUpdateAvailable: CheckAppUpdateAvailability
+
+  private val currentAppVersionCode = 1
+  private val diffToShowUpdate = 1
+
+  @Before
+  fun setup() {
+
+    val versionCodeCheck = { versionCode: Int, _: Application, _: AppUpdateConfig ->
+      versionCode.minus(currentAppVersionCode) >= diffToShowUpdate
+    }
+
+    checkUpdateAvailable = CheckAppUpdateAvailability(mock(), config, versionCodeCheck)
+  }
+
+  @Test
+  @Parameters(method = "params for checking app update")
+  fun `when app update is available and version is applicable, then shouldShowUpdateDialog should return true`(
+      availableVersionCode: Int,
+      updateAvailabilityState: Int,
+      flexibleUpdateIntent: PendingIntent?,
+      isInAppUpdateEnabled: Boolean,
+      shouldShow: Boolean
+  ) {
+    val packageName = "org.simple.clinic"
+    val immediateUpdateIntent = mock<PendingIntent>()
+
+    val appUpdateInfo = AppUpdateInfo(
+        packageName,
+        availableVersionCode,
+        updateAvailabilityState,
+        InstallStatus.DOWNLOADED,
+        immediateUpdateIntent,
+        flexibleUpdateIntent
+    )
+
+    whenever(config.inAppUpdateEnabled).thenReturn(isInAppUpdateEnabled)
+    whenever(config.numberOfVersionsToDiff).thenReturn(1)
+
+    val shouldShowDialog = checkUpdateAvailable.shouldShowUpdateDialog(appUpdateInfo)
+
+    assert(shouldShowDialog == shouldShow)
+  }
+
+  fun `params for checking app update`(): List<List<Any?>> {
+    return listOf(
```

Comment:
I generally prefer to create a local function named `testCase` so I can construct the instances with named parameters that are easier to understand. Do you feel that would be a good idea here? Otherwise, it's hard to tell what the different values mean.